docs/howto/plugins/actions.md

```diff
@@ -0,0 +1,479 @@
+    Make sure the workflow name is unique. We recommend using the
+    module path of the workflow class as workflow name to ensure uniqueness
+    among all the plugins added to a NOMAD installation.
+
+We define the workflow _function_ in the `run` method of the workflow
+definition class and use the Temporal decorator `workflow.run`. It describes the
+sequence of activities and the flow of data from one to another. Using appropriate data models, we pass the data from the workflow input to the activity inputs.
+
+Each activity is executed by `workflow.execute_activity` function which
+also specifies the activity's retry policy and timeouts.
+[Retry Policy](https://docs.temporal.io/encyclopedia/retry-policies)
+tells Temporal how to retry an activity that failed in the current execution.
+Attributes like `initial_interval`, `backoff_coefficient`, and
+`maximum_interval` control the interval between retries. The attribute
+`maximum_attempts` specifies the maximum retries that can be made in case of
+failures.
+
+Activity timeouts can detect failures, simply because the activity exceeds the
+maximum expected execution time. Temporal provides multiple
+[timeouts](https://docs.temporal.io/encyclopedia/detecting-activity-failures).
+The attribute `start_to_close_timeout` specifies the timeout for an activity
+execution, i.e., the time spent after a worker starts executing an activity
+till it is finished. For most cases, setting this alone is enough and
+recommended. Make sure that the timeout is longer than the maximum possible
+time for the activity execution to complete. For example, while setting one for
+an activity that makes an API call, determine the median call time and add some
+buffer to it.
+
+!!! tip "Important"
+    The default retry policy has unlimited `maximum_attempts`. We strongly
+    recommend to **always set a custom retry policy** with finite `maximum_attempts` to avoid forever running workflows.
+    In addition, **always set appropriate timeouts** for activities to prevent stuck executions.
+
+
+
+
+## Integrating action with schemas
+
+After actions are defined, it is possible to intergrate their workflows with
+[schemas](../../reference/glossary.md#schema) and run them from NOMAD
+entries.
+
+The workflows defined through `ActionHandler` have unique names associated with
+them. We can run a workflow using `start_workflow`
+function, which takes the workflow name, an instance of its input model, and
+the name of the task queue where the workflow run will be added:
+
+```py
+from nomad.orchestrator.utils import start_workflow
+from nomad.orchestrator.shared.constant import TaskQueue
+
+from nomad_example.actions.models import ExampleWorkflowInput
+
+workflow_id = start_workflow(
+    workflow_name='nomad_example.actions.workflows.ExampleWorkflow',
+    data=ExampleWorkflowInput(
+        user_id='NOMAD User ID',
+        upload_id='NOMAD Upload ID',
+        cid=962,  # CID for Water
+    ),
+    task_queue=TaskQueue.CPU,
+)
```

Comment:
- This should instead be called `start_action`.
 - Instead of `workflow_name`, you shold use `action_id`
 - One should not need to provide task_queue: it can be read from the action entry point configuration.